src/feedback/feedback.service.ts

```diff
@@ -46,27 +42,50 @@ export class FeedbackService {
 
   async createFeedback(creatorId: string, feedback: PostFeedbackDto) {
     const id = randomUUID();
-    // check if user can create feedback on this user
+    if (!this.haveDealWithUser(creatorId, feedback.userId)) {
+      throw new BadRequestException('This user can`t leave a feedback!');
+    }
     const newFeedback = {
       id,
       ...feedback,
       creatorId,
       created_date: new Date(),
     };
-    this.feedbacks.push(newFeedback);
+    const feedbackAdded = this.dataServices.feedbacks.create(newFeedback);
+    if (!feedbackAdded) {
+      throw new InternalServerErrorException();
+    }
     return newFeedback;
   }
 
-  async deleteFeedback(id: string, userId: string) {
-    const feedbackIdx = this.feedbacks.findIndex((el) => el.id === id);
-    if (feedbackIdx === -1) {
+  async deleteFeedback(id: string, creatorId: string) {
+    const feedback = await this.dataServices.feedbacks.getById(id);
+
+    if (!feedback) {
       throw new NotFoundException('No feedback with this id!');
     }
-    if (this.feedbacks[feedbackIdx].creatorId !== userId) {
+    if (feedback.creatorId !== creatorId) {
       throw new BadRequestException(
         'User can not delete feedback that is not his!',
       );
     }
-    return this.feedbacks.splice(feedbackIdx, 1)[0];
+
+    const deletedFeedback = await this.dataServices.feedbacks.remove(id);
+    if (!deletedFeedback) {
+      throw new InternalServerErrorException();
```

Comment:
Probably better not found exception because it will happen more often than dropped database